integration_tests/_helpers.py

```diff
@@ -397,6 +398,41 @@ def _start_span(
     ).start_span(span_name)
 
 
+class _DefaultAdminTokenSequestration:
```

Comment:
all of the object/method names feel a little tangential to what they do and I'm having a little bit of a hard time following. In this docstring you're describing a broader strategy but the class seems to be doing something much more specific. Are we storing away admin tokens here so they can't be used if we ever see them? `intersect` as a method name feels opaque to me as well—is the point to check whether or not we've marked a specific token as an admin token? If so, could we use `is_sequestered_token` or `is_admin_token`?

Reply:
Maybe the logic for whether or not it's an admin token can live entirely in here so it's a little bit more clear *how* this class should be used

Reply:
> Are we storing away admin tokens here so they can't be used if we ever see them

This class is specifically for storing tokens: it doesn't dictate whether they can or cannot be used again.

Reply:
> In this docstring you're describing a broader strategy but the class seems to be doing something much more specific

the docstring is describing the why we need this class, but this class is just one piece of the strategy

Reply:
> Maybe the logic for whether or not it's an admin token can live entirely in here

sure, although that seems to be more than "single responsibility".

Reply:
> intersect as a method name feels opaque to me as well—is the point to check whether or not we've marked a specific token as an admin token

it just to check whether it's been sequestered. it makes no claims about the what or why?

Reply:
I would argue that the single responsibility might be: "check to see if this token is usable" because I don't have to understand anything deeply about concurrency race conditions to figure out how it fits into the test helpers.

I think if I were to only look at this object as some kind of temporary stash, I'd have to really understand carefully how all the routes interact/work to know if the implementation below is correct